src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -129,7 +129,7 @@ public int order() {
                 threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN, Origin.LOCAL.toString());
             }
             
-            if(complianceConfig != null && complianceConfig.isEnabled()) {
+            if(auditLog.getConfig() != null && auditLog.getConfig().isEnabled()) {
```

Comment:
`auditLog.getConfig()` gets used a lot. Maybe assign this value to a member variable?

Reply:
`auditLog.getConfig()` cannot be assigned to a member variable because it must always fetch the latest configuration that AuditLog service has (because of hot-reloading of configuration) when function is executed. 
However usage within the same function can be simplified.

Reply:
Hot-reloading is not implemented yet. 
I will clean this up and subscribe all classes to change listeners that need to be notified in the next hot-reloading PR